packages/react-charting/src/utilities/utilities.ts

```diff
@@ -303,7 +309,57 @@ export function prepareDatapoints(maxVal: number, minVal: number, splitInto: num
  * @param {IYAxisParams} yAxisParams
  * @param {boolean} isRtl
  */
-export function createYAxis(yAxisParams: IYAxisParams, isRtl: boolean, axisData: IAxisData) {
+export function createYAxis(
+  yAxisParams: IYAxisParams,
+  isRtl: boolean,
+  axisData: IAxisData,
+  chartType: ChartTypes,
+  barWidth: number,
+) {
+  switch (chartType) {
+    case ChartTypes.HorizontalBarChartWithAxis:
+      return createYAxisForHorizontalBarChartWithAxis(yAxisParams, isRtl, axisData, barWidth!);
+    default:
+      return createYAxisForOtherCharts(yAxisParams, isRtl, axisData);
+  }
+}
+
+export function createYAxisForHorizontalBarChartWithAxis(
+  yAxisParams: IYAxisParams,
+  isRtl: boolean,
+  axisData: IAxisData,
+  barWidth: number,
+) {
+  const {
+    yMinMaxValues = { startValue: 0, endValue: 0 },
+    yAxisElement = null,
+    yMaxValue = 0,
+    yMinValue = 0,
+    containerHeight,
+    margins,
+    tickPadding = 12,
+    maxOfYVal = 0,
+    yAxisTickFormat,
+    yAxisTickCount = 4,
+    eventAnnotationProps,
```

Comment:
we dont have eventAnnotationProps in this chart type

Reply:
This is removed.